File: compiler/bench/licm2.fs
\ expected: 100000000
\ Hoist computation out of loop

: main
  3 7 +   \ loop-invariant computation = 10
  0 swap  \ sum, invariant
  100000000 0 do
    dup 10 = if swap 1+ swap then
  loop drop . cr ;

File: compiler/bench/licm2.c
// expected: 100000000
// Hoist computation out of loop
#include <stdio.h>

int main() {
    long a = 3, b = 7;
    long sum = 0;
    for (long i = 0; i < 100000000; i++) {
        long x = a + b;  // loop invariant
        if (x == 10) sum++;
    }
    printf("%ld\n", sum);
    return 0;
}
